packages/ember-views/lib/system/utils.js

```diff
@@ -88,18 +88,25 @@ export function getChildViews(view) {
 }
 
 export function initChildViews(view) {
-  view[CHILD_VIEW_IDS] = [];
+  let childViews = [];
+  CHILD_VIEW_IDS.set(view, childViews);
+  return childViews;
 }
 
 export function addChildView(parent, child) {
-  parent[CHILD_VIEW_IDS].push(getViewId(child));
+  let childViews = CHILD_VIEW_IDS.get(parent);
+  if (childViews === undefined) {
+    childViews = initChildViews(parent);
+  }
+
+  childViews.push(getViewId(child));
 }
 
 export function collectChildViews(view, registry) {
   let ids = [];
   let views = [];
 
-  view[CHILD_VIEW_IDS].forEach(id => {
+  (CHILD_VIEW_IDS.get(view) || []).forEach(id => {
```

Comment:
Seems like this should be split up on multiple lines, and in the case where `CHILD_VIEW_IDS.get(view)` is undefined we can return early (e.g. no need for the `forEach`)...